entanglement_forging/core/wrappers/entanglement_forged_driver.py

```diff
@@ -48,26 +64,66 @@ def __init__(
         self._num_beta = num_beta
         self._nuclear_repulsion_energy = nuclear_repulsion_energy
 
-    def run(self) -> QMolecule:
+    def run(self) -> ElectronicStructureDriverResult:
         """Returns QMolecule constructed from input data."""
-        q_molecule = QMolecule()
-        q_molecule.atom_symbol = []
-        q_molecule.atom_xyz = []
-        q_molecule.hcore = self._hcore
-        q_molecule.mo_coeff = self._mo_coeff
-        q_molecule.eri = self._eri
-        q_molecule.num_molecular_orbitals = self._mo_coeff.shape[0]
-        q_molecule.num_alpha = self._num_alpha
-        q_molecule.num_beta = self._num_beta
-        q_molecule.nuclear_repulsion_energy = self._nuclear_repulsion_energy
+        # Create ParticleNumber property
+        particle_number = ParticleNumber(
+            self._mo_coeff.shape[0], (self._num_alpha, self._num_beta)
+        )
+
+        # Convert one body integrals from AO to MO basis
+        one_body_in_mo_basis = np.dot(
+            np.dot(np.transpose(self._mo_coeff), self._hcore), self._mo_coeff
+        )
+
+        # Convert two-body integrals from AO to MO basis
+        dim = self._eri.shape[0]
+        two_body_in_mo_basis = np.zeros((dim, dim, dim, dim))
+        for a_i in range(dim):
+            temp1 = np.einsum("i,i...->...", self._mo_coeff[:, a_i], self._eri)
+            for b_i in range(dim):
+                temp2 = np.einsum("j,j...->...", self._mo_coeff[:, b_i], temp1)
+                temp3 = np.einsum("kc,k...->...c", self._mo_coeff, temp2)
+                two_body_in_mo_basis[a_i, b_i, :, :] = np.einsum(
+                    "ld,l...c->...cd", self._mo_coeff, temp3
+                )
```

Comment:
You can let the `properties` framework do these transformatiions for you. I'll add suggestions below on how to do that, which will allow you to remove all of this :+1:
See for example how the `PySCFDriver` is doing this:
https://github.com/Qiskit/qiskit-nature/blob/aae9c01b4d02bf5dbaa342c757be2cfef60a31d5/qiskit_nature/drivers/second_quantization/pyscfd/pyscfdriver.py#L620-L639